lib/addressable/uri.rb

```diff
@@ -851,7 +849,8 @@ def initialize(options={})
         self.query_values = options[:query_values] if options[:query_values]
         self.fragment = options[:fragment] if options[:fragment]
       end
-      self.to_s
+
+      to_s # enforce path validation
```

Comment:
this line took me a bit to figure out, so I figured it was comment worthy.    based on the specs, it looks like `initialize` leverages the validation of `self.path` done in `.to_s`.  but moving it into `.validate` causes other test failures...